### ISSUE TITLE
Loading fails on STM32F051-R8T6

### ISSUE THREAD

**Issue by ghost** [texane/stlink#153]
I use st-util with a GCC toolchain on eclipse and when I try to load the program to the board (STM32F0 Discovery) it fails. I have to unplug the USB cable then restart the uploading and after that it works perfectly. I used to use st-util with STM32F103/107 devices without any problem so it should be a compatibility issue with this specific hardware. Here is what I get when it fails : 

```
2013-05-17T14:44:58 INFO src/stlink-common.c: Loading device parameters....
2013-05-17T14:44:58 INFO src/stlink-common.c: Device connected is: F0 device, id 0x20006440
2013-05-17T14:44:58 INFO src/stlink-common.c: SRAM size: 0x2000 bytes (8 KiB), Flash: 0x8000 bytes (32 KiB) in pages of 1024 bytes
2013-05-17T14:45:01 INFO src/stlink-common.c: Attempting to write 1024 (0x400) bytes to stm32 address: 134217728 (0x8000000)
Chip ID is 00000440, Core ID is  0bb11477.
KARL - should read back as 0x03, not 60 02 00 00
init watchpoints
Listening at *:4200...
GDB connected.
recv: qSupported:qRelocInsn+
query: Supported;qRelocInsn+
send: PacketSize=3fff;qXfer:memory-map:read+
recv: Hg0
send: 
recv: ?
send: S05
recv: Hc-1
send: 
recv: qC
send: 
recv: qAttached
query: Attached;
send: 
recv: qOffsets
query: Offsets;
send: 
recv: g
send: 00000000dc000020dc0000200a000000e80100200a0000000044014001000000004801402f000000ffffffffffffffff03000000100f00201953000836370008
recv: qXfer:memory-map:read::0,fff
query: Xfer;memory-map:read::0,fff
Xfer: type:memory-map;op:read;annex:;addr:0;length:4095
send: m<?xml version="1.0"?><!DOCTYPE memory-map PUBLIC "+//IDN gnu.org//DTD GDB Memory Map V1.0//EN"     "http://sourceware.org/gdb/gdb-memory-map.dtd"><memory-map>  <memory type="rom" start="0x00000000" length="0x8000"/>  <memory type="ram" start="0x20000000" length="0x2000"/>  <memory type="flash" start="0x08000000" length="0x8000">    <property name="blocksize">0x400</property>  </memory>  <memory type="ram" start="0x40000000" length="0x1fffffff"/>  <memory type="ram" start="0xe0000000" length="0x1fffffff"/>  <memory type="rom" start="0x1fffec00" length="0xc00"/>  <memory type="rom" start="0x1ffff800" length="0x10"/></memory-map>
recv: qXfer:memory-map:read::27a,d85
query: Xfer;memory-map:read::27a,d85
Xfer: type:memory-map;op:read;annex:;addr:634;length:3461
send: l
recv: m8003736,4
send: 102df9d1
recv: m20000f00,40
send: 030000001953000838370008250000810200000015000000540f0020533f000800e100e0f9ffffff00440140010000000000000000e100e003000000f9380008
recv: qTStatus
query: TStatus;
send: 
recv: qfThreadInfo
query: fThreadInfo;
send: 
recv: qL1200000000000000000
send: 
recv: qRcmd,72657365742068616c74
query: Rcmd,72657365742068616c74;
KARL - should read back as 0x03, not 60 02 00 00
init watchpoints
Rcmd: reset
send: OK
recv: vFlashErase:08000000,00005c00
FlashErase: addr:08000000,len:5c00
send: OK
recv: vFlashWrite:8000000:
binary packet 196 -> 196
send: OK
recv: vFlashWrite:80000c8:µL}x
binary packet 16347 -> 16104
send: OK
recv: vFlashWrite:8003fb0:h    êI
h
íKhêI
h
íKhê
binary packet 6590 -> 6488
send: OK
recv: vFlashWrite:8005908:Ò
binary packet 4 -> 4
send: OK
recv: vFlashWrite:800590c:…
binary packet 4 -> 4
send: OK
recv: vFlashWrite:8005910:
binary packet 316 -> 316
send: OK
recv: vFlashDone
flash_do: block 08000000 -> 5c00
flash_do: page 08000000

Flash page at addr: 0x08000000 erased2013-05-17T14:45:01 INFO src/stlink-common.c: Finished erasing 1 pages of 1024 (0x400) bytes
2013-05-17T14:45:01 INFO src/stlink-common.c: Starting Flash write for VL/F0 core id
2013-05-17T14:45:01 INFO src/stlink-common.c: Successfully loaded flash loader in sram
2013-05-17T14:45:09 ERROR src/stlink-common.c: flash loader run error
2013-05-17T14:45:09 ERROR src/stlink-common.c: run_flash_loader(0x8000000) failed! == -1
cannot recv: -2

send: E00
recv: qL1200000000000000000
send: 
recv: m20000f00,40
send: 030000001953000800000020250000210200000015000000540f0020533f000800e100e0f9ffffff00440140010000000000000000e100e003000000f9380008
```

**Comment by xor-gate** [comment]
Could this be verified if it works now with `v1.2.0` or from `master` ?

**Comment by xor-gate** [comment]
Closing due to inactivity of original issuer. Feel free to open a new issue if the problem exists with `v1.2.0` or current `master`. Thank you for your contribution.

**Comment by Nightwalker-87** [comment]
This still remains unverified.

**Comment by Nightwalker-87** [comment]
@bacam and @lestofante: May I ask you for some help to verify this is working since release v1.2.0? I am currently reviewing old issues and leftovers to ensure proper solutions have been found and that tickets can be allocated to the correct milestones.

**Comment by bacam** [comment]
Unfortunately my board is in a building that's closed for the foreseeable future.

**Comment by Nightwalker-87** [comment]
Thx for replying. No problem. We can come back to it at a later time should there be no alternative to it.

**Comment by Nightwalker-87** [comment]
Fixed in commit 079a154f5f78c66a16653e62cfdf8674d9b34d90 which added support for this board type.

**Comment by Nightwalker-87** [comment]
@bacam: No further help needed anymore, as I found the related fix.
The problem of **lestofante** is something different not related to this topic.
